consensus/cometbft/service/encoding/encoding.go

```diff
@@ -81,7 +83,14 @@ func UnmarshalBeaconBlockFromABCIRequest(
 		return signedBlk, ErrNilBeaconBlockInRequest
 	}
 
-	return ctypes.NewSignedBeaconBlockFromSSZ(blkBz, forkVersion)
+	block, err := ctypes.NewEmptySignedBeaconBlockWithVersion(forkVersion)
+	if err != nil {
+		return nil, fmt.Errorf("attempt at building block with wrong version %s: %w", forkVersion, err)
+	}
+	if err = decoder.SSZUnmarshal(blkBz, block); err != nil {
+		return nil, err
+	}
+	return block, nil
```

Comment:
just noting, I would avoid making functional/behavior changes in this PR. i.e. there is no non-nil verification of withdrawals as there was before

Reply:
Indeed this changed a bit and I should have explained this better. I think the behaviour is effectively the same.
`Withdrawals` are instantiated in `NewEmptyExecutionPayloadWithVersion` (only if the fork is high enough) and the check is done in `ValidateAfterDecodingSSZ` which is called "recursively" from signed beacon block down